Pr/ios sdk docs (#926)
clerk/clerk-docs: docs/quickstarts/ios.mdx

---
title: Use Clerk with iOS
description: Learn how to use Clerk to quickly and easily add secure authentication and user management to your iOS application.
---

# iOS Quickstart (Beta)

<TutorialHero
  framework="ios"
  beforeYouStart={[
    {
      title: "Set up a Clerk application",
      link: "https://clerk.com/docs/quickstarts/setup-clerk",
    },
  ]}
>

- Add the Clerk iOS SDK to your application
- Sign in an existing user
- Access information about your user

</TutorialHero>

> Learn how to use Clerk to quickly and easily add secure authentication and user management to your iOS application.

Reviewer: ```suggestion
Learn how to use Clerk to add secure authentication and user management to your iOS application.
```

<Steps>

### Install the SDK

##### Using Swift Package Manager (SPM)

> Open the following menu item in Xcode:

Reviewer: 1. In Xcode, select ** File > Add Package Dependencies...**

> 

Reviewer: ```suggestion
```

> File > Add Package Dependencies...

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

> In the Search or Enter Package URL search box enter this URL:

Reviewer: ```suggestion
In the **Search or Enter Package URL** field, enter the following URL:
```

```swift
https://github.com/clerk/clerk-ios
```

> Then, select your preferred dependency rule and press Add Package.

Reviewer: ```suggestion
Then, select your preferred dependency rule and select **Add Package**.
```

> <Callout>

Reviewer: ```suggestion
<Callout type="info">
```

For more information on SPM, visit the [official documentation](https://developer.apple.com/documentation/xcode/adding-package-dependencies-to-your-app).
</Callout>

### Configure and Load Clerk

> 

Reviewer: ```suggestion

To configure Clerk in your local project, you first need your publishable API key:

<SignedIn>

<Callout type="info">
  Because you're signed in, the code snippet below will contain the publishable key for your most recently active project. Unless you want to use a different project, you can skip these steps.
</Callout>

</SignedIn>

1. Navigate to the Clerk Dashboard.
2. In the navigation sidebar, select [API Keys](https://dashboard.clerk.com/last-active?path=api-keys).
3. In the **Quick Copy** section, copy your Clerk publishable key.
4. Add the following code snippet to your project, assigning your publishable key to the `publishableKey` property:
```
Adds instructions for getting the API key

<InjectKeys>

<Tabs type="framework" items={["SwiftUI", "UIKit"]}>

<Tab>
```swift filename="App.swift"
import SwiftUI
import ClerkSDK

@main
struct ClerkApp: App {
    var body: some Scene {
        WindowGroup {
            SignInView()
                .task {
                    Clerk.shared.configure(publishableKey: {{pub_key}})
                    try? await Clerk.shared.load()
                }
        }
    }
}
```
</Tab>

<Tab>
```swift filename="AppDelegate"
import UIKit
import ClerkSDK

func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
    Clerk.shared.configure(publishableKey: {{pub_key}})
    Task { try? await Clerk.shared.load() }
    return true
}
```
</Tab>

</Tabs>

> ### Create a Sign In

Reviewer: ```suggestion
### Create a sign-in view
```

> 

Reviewer: ```suggestion

To see Clerk in action, create a view that allows visitors to sign-in to your Clerk app:
```
